src/components/navigation/nav-tab-panel.tsx

```diff
@@ -184,4 +185,13 @@ export class NavTabPanel extends BaseComponent<IProps> {
     const { persistentUI } = this.stores;
     persistentUI.setDividerPosition(kDividerMin);
   };
+
+  private hideChatRules: Array<(tabIndex: number, userType: string) => boolean> = [
+    // Hide chat for students on the problems tab
+    (tabIndex, userType) => userType === "student" && tabIndex === 0
```

Comment:
This is good enough for now. But I think its possible the problems tab can be located other than at tabIndex 0.  If possible what might be better is if chat panel visibility would be based on the focus document type. I don't think it is worth fixing though, this is good enough.